### PR TITLE
[Docs] Add Aggregations list to the high-level client documentation

### PR DESCRIPTION
**Pull request by tlrx**
Related to https://github.com/elastic/elasticsearch/pull/25984#pullrequestreview-53710901

### CODE REVIEW

File: docs/Versions.asciidoc [hunk 1]
```diff
@@ -45,6 +46,7 @@ ifeval::["{release-state}"!="unreleased"]
 :painless-javadoc: https://artifacts.elastic.co/javadoc/org/elasticsearch/painless/lang-painless/{version}
 :parent-join-client-javadoc: https://artifacts.elastic.co/javadoc/org/elasticsearch/plugin/parent-join-client/{version}
 :percolator-client-javadoc: https://artifacts.elastic.co/javadoc/org/elasticsearch/plugin/percolator-client/{version}
+:matrixstats-client-javadoc: https://artifacts.elastic.co/javadoc/org/elasticsearch/plugin/aggs-matrix-stats-client/{version}-SNAPSHOT
```
Comment (javanna) [inline docs/Versions.asciidoc]: remove the -SNAPSHOT suffix at the end ?

File: docs/java-rest/high-level/apis/aggs.asciidoc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+This page lists all the available aggregations with their corresponding `AggregationBuilder` class name and helper method name in the
+`AggregationBuilders` or `PipelineAggregatorBuilders` utility classes.
+
+//:agg-ref:        {elasticsearch-javadoc}/org/elasticsearch/search/aggregations
+//:parentjoin-ref: {parent-join-client-javadoc}/org/elasticsearch/join/aggregations
+//:matrixstats-ref: {matrixstats-client-javadoc}/org/elasticsearch/search/aggregations
+
+:agg-ref:        https://snapshots.elastic.co/javadoc/org/elasticsearch/elasticsearch/5.6.0-SNAPSHOT/org/elasticsearch/search/aggregations
+:parentjoin-ref: https://snapshots.elastic.co/javadoc/org/elasticsearch/plugin/parent-join-client/5.6.0-SNAPSHOT/org/elasticsearch/join/aggregations
+:matrixstats-ref: https://snapshots.elastic.co/javadoc/org/elasticsearch/plugin/aggs-matrix-stats-client/5.6.0-SNAPSHOT/org/elasticsearch/search/aggregations
```
Comment (javanna) [inline docs/java-rest/high-level/apis/aggs.asciidoc]: versions are hardcoded here? they should not be I think

**Review by javanna — APPROVED** [review]
left a couple of comments on versions, LGTM otherwise

File: docs/Versions.asciidoc (hunk 1 above)
Comment (tlrx) [inline docs/Versions.asciidoc]: Good catch

File: docs/java-rest/high-level/apis/aggs.asciidoc (hunk 2 above)
Comment (tlrx) [inline docs/java-rest/high-level/apis/aggs.asciidoc]: Sorry that's a leftover: I used hardcoded attributes to tests all the generated links. The right values are commented few lines above:

:agg-ref:        {elasticsearch-javadoc}/org/elasticsearch/search/aggregations
:parentjoin-ref: {parent-join-client-javadoc}/org/elasticsearch/join/aggregations
:matrixstats-ref: {matrixstats-client-javadoc}/org/elasticsearch/search/aggregations

**Comment by tlrx** [comment]
Thanks @javanna !

**Pull request merged**